fastlane_bot/events/exchanges/solidly_v2.py

```diff
@@ -41,31 +49,49 @@ def add_pool(self, pool: Pool):
         self.pools[pool.state["address"]] = pool
 
     def get_abi(self):
-        return VELOCIMETER_V2_POOL_ABI if "velocimeter" in self.exchange_name else SOLIDLY_V2_POOL_ABI
+        try:
+            return EXCHANGE_INFO.get(self.exchange_name).get("pool_abi")
+        except KeyError:
+            raise self.WrongExchangeException(f"[exchanges/solidly_v2 get_Abi] Exchange {self.exchange_name} not in supported exchanges.")
 
     @property
     def get_factory_abi(self):
-        return VELOCIMETER_V2_FACTORY_ABI if "velocimeter" in self.exchange_name else SOLIDLY_V2_FACTORY_ABI
+        try:
+            return EXCHANGE_INFO.get(self.exchange_name).get("factory_abi")
+        except KeyError:
+            raise self.WrongExchangeException(f"[exchanges/solidly_v2 get_Abi] Exchange {self.exchange_name} not in supported exchanges.")
 
     def get_events(self, contract: Contract) -> List[Type[Contract]]:
         return [contract.events.Sync] if self.exchange_initialized else []
 
     async def get_fee(self, address: str, contract: Contract) -> Tuple[str, float]:
-        if "velocimeter" in self.exchange_name:
+        if self.exchange_name == "velocimeter_v2":
             default_fee = await self.factory_contract.caller.getFee(address)
-            default_fee = float(default_fee) / 10000
-
-        elif self.exchange_name in ["scale_v2", "equalizer_v2"]:
+        elif self.exchange_name == "equalizer_v2":
             default_fee = await self.factory_contract.caller.getRealFee(address)
-            default_fee = float(default_fee) / 10 ** 18 if self.exchange_name == "scale_v2" else float(default_fee) / 10000
-        else:
+        elif self.exchange_name == "scale_v2":
+            default_fee = await self.factory_contract.caller.getRealFee(address)
+        elif self.exchange_name == "aerodrome_v2":
             is_stable = await contract.caller.stable()
             default_fee = await self.factory_contract.caller.getFee(address, is_stable)
-            default_fee = float(default_fee) / 10000
-        return str(default_fee), default_fee
+        elif self.exchange_name == "velodrome_v2":
+            is_stable = await contract.caller.stable()
+            default_fee = await self.factory_contract.caller.getFee(address, is_stable)
+        elif self.exchange_name == "solidly_v2":
+            is_stable = await contract.caller.stable()
+            default_fee = await self.factory_contract.caller.getFee(address, is_stable)
+        else:
+            raise self.WrongExchangeException(f"[exchanges/solidly_v2 get_Abi] Exchange {self.exchange_name} not in supported exchanges.")
+
+        fee_float = float(default_fee) / 10 ** EXCHANGE_INFO.get(self.exchange_name).get("decimals")
```

Comment:
In order to use `EXCHANGE_INFO` here in the same way as above, implement the following two global functions:
```
async def getFee1(address: str, contract: Contract, factory_contract: Contract):
    return await factory_contract.caller.getFee(address)

async def getFee2(address: str, contract: Contract, factory_contract: Contract):
    return await factory_contract.caller. getRealFee(address)

async def getFee3(address: str, contract: Contract, factory_contract: Contract):
    return await self.factory_contract.caller.getFee(address, await contract.caller.stable())
```
Then add a `"get_fee"` key in each one of the values on the `EXCHANGE_INFO` dictionary, with the value of that key being the appropriate function out of these 3 functions above.

Reply:
Updated